### ISSUE TITLE
Add common repetitions

### ISSUE THREAD

**Issue by korverdev** [codespell-project/codespell#3160]
While not strictly spelling, I've noticed a few common words mistakenly repeated in a few of the projects I work on such as "the the". [A potential starting point for this is the Wikipedia list of common repetitions.](https://en.wikipedia.org/wiki/Wikipedia:Lists_of_common_misspellings/Repetitions)

I'm open to tackling this, but wanted to get maintainer thoughts on whether this is in-scope for this project and in which dictionary these common repetitions should be placed or if they justify their own `.txt` file.

**Comment by DimitriPapadopoulos** [comment]
That would be a nice addition, I don't know if it's worth the additional complexity. @larsoner?

**Comment by larsoner** [comment]
Agreed it's potentially worth it. My concern would mostly be with increased computation time. For example, if checking for repeats doubles `codespell` execution time, that's not trivial. We'd need to address that for example with a `--repeats` opt-in or something.

Regarding where the repeats should live I think it should be their own `.txt` file. I think we currently prohibit spaces on the left side of the `.txt` files, so it could make sense to make the repeats files explicitly have `the the->the` for example, i.e., every error *must* have a space (rather than no entry can have a space). It's a bit redundant but it's also fairly explicit, and makes it clear from the file contents whether it's a misspelling txt or repeats txt. It also opens the possibility of eventually doing fixes for incorrect short phrases (not sure if we want to go that route but this would leave open the possibility).

**Comment by peternewman** [comment]
Isn't the main issue that we currently extract words by splitting on word boundaries using `\b`. To find duplicates we'd need to either change the regex or pass in pairs of words.

Note that spellintian for example throws false positives in code with e.g. if if foo then bar else baz else bak type constructs. You also need to deal with line wrapping to catch all instances, although just catching some would be a start.
